src/database-layer/db.ts

```diff
@@ -707,9 +714,55 @@ if (maybePg != null) {
 				`);
 			}
 		}
+
+		let listenerClient: Pg.PoolClient | null = null;
+		async function listen(
+			channel: string,
+			fn: (...args: any[]) => Promise<void>,
+		) {
+			while (listenerClient == null) {
```

Comment:
This means that it's only possible to add a listener whilst `listenerClient == null`, ie just once as the first listener (unless you hit the race condition mentioned below)